uuid-cluster-dns/README.md

```diff
@@ -0,0 +1,49 @@
+# Using UUIDs for cluster endpoint domains
+
+The current approach for building cluster domains is to include the workload cluster's ID as a subdomain of the management cluster it belongs to, e.g.:
+
+```
+https://api.j9yr2.k8s.gauss.eu-west-1.aws.gigantic.io
+```
+
+This approach has a few drawbacks:
+
+1. The workload cluster is tied to the management cluster, making migration messy
+2. Workload clusters cannot be "promoted" to management clusters as the domain would no longer make sense
+3. Several customers have their own domain set up, leading to inconsistency between MCs
+4. The provider (and region) is baked into the domain
+
+An alternative "flat" DNS structure has been proposed that removes the link between WC and MC. E.g.:
+
+```
+https://api.j9yr2.k8s.[customer name].gigantic.io
+```
+
+While this solves some of the above drawbacks (1, 2, 4) it introduces its own complications:
+
+1. The customer name (or codename) is exposed in the domain
+2. There is no enforced global uniqueness on cluster IDs so if a customer has multiple MCs there is a chance that two workload clusters have the same ID, leading to conflicting domains
```

Comment:
I think this is a problem but to me it reveals a recurring issue:
We wanted `cluster IDs` to be a unique and "random" identifier. But users wanted to be able to control what it would look like and add meaning to them. Also to make it easier for automation to predict the ID.

Now we are introducing another random identifier - how can we prevent the same issues from reappearing?

Reply:
Isn't that what the `kubectl-gs template` command is for?

> But users wanted to be able to control what it would look like and add meaning to them.

This is the purpose of the description annotation isn't it?

Reply:
Still customers choose IDs such as `peu01` as a specific means to give them reason.

Reply:
There's no reason that names / IDs can't still be a thing. Just under the hood every cluster is given a UUID to be used for things that really need to be unique like DNS.

Reply:
In fact, based on what @marians said it might actually be beneficial to have them as different things.

Reply:
I think for UX purposes we can still have names/IDs, and UUIDs are then only for DNS. AFAICS customers do not really look at the DNS of their APIs, but rather care about the context name in the kubeconfig, but we can verify that assumption.

Reply:
Yeah, bottom line: DNS and cluster name (as an identifier) are separate things.